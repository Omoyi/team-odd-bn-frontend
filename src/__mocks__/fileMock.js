const mockData = {
  errorData: {
    error: 'I am feke error',
    hideErrors: jest.fn(),
  },
  signinDummyDataWithErrors: {
    socialAuth: false,
    errors: 'I am dummy',
    buttonName: 'log in with Google',
  },
  destructLocationData: {
    location: {
      search: 'https://dummy.example.url/?info=Iamtoken',
    },
  },
  socialAuthDummyData: {
    authUserData: jest.fn(),
    location: {
      search: 'https://dummy.example.url/?info=Iamtoken',
    },
  },
  socialAuthReducer: {
    type: 'SOCIAL_AUTH_SUCCESS',
    payload: 'successfuly authenticate',
  },
  errorActionReducer: {
    type: 'ACTION_FAIL_ERRORS',
    payload: 'I am error',
  },
  profileProps: {
    profile: {
      data: {
        user: {
          firstName: 'Ivy',
          lastName: 'League',
        },
        imageURL: 'image.png',
      },
    },
    viewProfile: {
      profile: {
        data: {
          user: {
            firstName: 'Ivy',
            lastName: 'League',
          },
          imageURL: 'image.png',
        },
      },
    },
    profileError: {},
    message: 'success',
    getProfile: jest.fn(),
  },

  myProfileProps: {
    profile: {
      data: {
        user: {
          firstName: 'Ivy',
          lastName: 'League',
        },
        gender: 'female',
        birthDate: '01/01/1990',
        address: 'Kigali',
        department: 'Finance',
        imageURL: 'image.png',
        managerId: 6,
        bio: 'Work hard play hard ',
      },
    },
    profileError: {},
    message: 'success',
    updateProfileError: {},
    imageURL: {},
    imageURLError: {},
    getProfile: jest.fn(),
    updateProfile: jest.fn(),
    uploadImage: jest.fn(),
  },
  props: {
    spinner: true,
    profile: {
      data: {
        user: {
          firstName: 'Ivy',
          lastName: 'League',
        },
        gender: 'female',
        birthDate: '01/01/1990',
        address: 'Kigali',
        department: 'Finance',
        imageURL: 'image.png',
        managerId: 6,
        bio: 'Work hard play hard ',
      },
    },
    users: {
      data: [
        {
          user: {
            firstName: 'Ivy',
            lastName: 'League',
          },
          gender: 'female',
          birthDate: '01/01/1990',
          address: 'Kigali',
          department: 'Finance',
          imageURL: 'image.png',
          managerId: 6,
          roleId: 6,
          bio: 'Work hard play hard ',
        },
        {
          user: {
            firstName: 'Ivy',
            lastName: 'League',
          },
          gender: 'female',
          birthDate: '01/01/1990',
          address: 'Kigali',
          department: 'Finance',
          imageURL: 'image.png',
          managerId: 6,
          roleId: 6,
          bio: 'Work hard play hard ',
        },
      ],
    },
    profileError: {},
    message: 'success',
    updateProfileError: {},
    imageURL: {},
    imageURLError: {},
    getProfile: jest.fn(),
    updateProfile: jest.fn(),
    uploadImage: jest.fn(),
    getManagers: jest.fn(),
    updateSpinnerStatus: jest.fn(),
    handleFiles: jest.fn(),
  },

  emptyProfile: {
    data: {
      user: {
        firstName: null,
        lastName: null,
      },
      gender: null,
      birthDate: null,
      address: null,
      department: null,
      imageURL: null,
      managerId: null,
      bio: null,
    },
  },
  Navbarprops: {
    profile: {
      data: {
        user: {
          firstName: 'Ivy',
          lastName: 'League',
        },
        imageURL: 'image.png',
      },
      profileError: null,
      message: null,
    },
  },
  viewProfile: {
    profile: null,
    profileError: null,
    updateProfileError: null,
    imageURL: null,
    imageURLError: null,
  },
  allUsers: {
    users: null,
    usersError: null,
  },
  updateProfile: {
    message: null,
    formErrors: {
      address: '',
      department: '',
      bio: '',
    },
  },
  navbarMainState: {
    viewProfile: {
      profile: null,
      profileError: null,
      message: null,
    },
  },
  navbarInitialState: {
    viewProfile: {
      profile: null,
      profileError: null,
    },
  },
  profileMainState: {
    viewProfile: {
      spinner: true,
      profile: null,
      profileError: null,
      message: null,
    },
  },
  signoutProps: {
    signout: jest.fn(),
    isSignout: true,
    mapStateToProps: jest.fn(),
    mapDispatchToProps: jest.fn(),
    signoutAction: jest.fn(),
    state: {
      signout: true,
      isLoggedin: {},
    },
    signoutSuccessfully: {
      message: 'Loggout successfully',
    },
    signoutFail: {
      message: 'Loggout failed',
    },
    successReducer: {
      type: 'SIGNOUT_SUCCESS',
      payload: true,
    },
    failureReducer: {
      type: 'SIGNOUT_FAILURE',
      payload: true,
    },
    logout: jest.fn(),
    history: {
      push: jest.fn(),
    },
    signoutUser: jest.fn(),
  },
  sidebarProps: {
    click: jest.fn(),
    backdropClickHandler: jest.fn(),
    drawerToggleClickHandler: jest.fn(),
    drawerClickHandler: jest.fn(),
    mapStateToProps: jest.fn(),
    mapDispatchToProps: jest.fn(),
    show: true,
    sideDrawerOpen: true,
    image: 'image.jpg',
    firstName: 'John',
    lastName: 'Doe',
    profile: {
      viewProfile: {
        profile: 'data profile',
      },
      data: {
        managerId: 1,
      },
    },
    profileError: 'profileError',
    state: {
      viewProfile: {
        profile: 'data',
      },
      profileError: 'error',
      sideDrawerOpen: 'function',
      children: 'children component',
    },
    prevProps: {
      profile: {
        data: {
          imageURL: 'Iamge.jpeg',
          managerId: 1,
          user: {
            firstName: 'John',
            lastName: 'Doe',
          },
        },
      },
    },
    UNSAFE_componentWillReceiveProps: jest.fn(),
    getCurrentUserinfo: jest.fn(),
    signoutUser: jest.fn(),
    history: {
      push: jest.fn(),
    },
    logout: jest.fn(),
  },
  userProfileMockData: {
    spinnerStatusAction: (data) => ({
      type: 'SPINNER_STATUS',
      spinner: data,
    }),
    successfulyProfileReducer: (data) => ({
      type: 'FETCH_PROFILE_SUCCESS',
      payload: data,
    }),
    failedProfileReducer: (data) => ({
      type: 'FETCH_PROFILE_ERROR',
      payload: data,
    }),
    successResponse: {
      profileInformation: {
        data: {
          firstName: 'John',
          lastName: 'Doe',
          img: 'dummy.image',
        },
      },
    },
    errorResponse: {
      error: {
        response: {
          data: {
            info: 'No profile',
          },
        },
      },
    },
    fetchUsersSuccessAction: (profileInformation) => ({
      type: 'FETCH_PROFILE_SUCCESS',
      payload: profileInformation.data,
    }),
    fetchUsersFailure: (error) => ({
      type: 'FETCH_PROFILE_ERROR',
      payload: error.response.data,
    }),
    userProfileMocData: {
      getProfile: jest.fn(),
      updateSpinnerStatus: jest.fn(),
    },
    roundTripProps: {
      spinner: true,
      roundTrip: {
        data: {
          originId: 1,
          destinationId: 1,
          reason: 'Meeting department heads',
          startDate: '2020-12-02',
          returnDate: '2030-02-03',
        },
        getCities: jest.fn(),
        createRoundTrip: jest.fn(),
        updateSpinnerStatus: jest.fn(),
      },
    },
    roundTripMainState: {
      trips: {
        roundTrip: {
          roundTripMessage: null,
          roundTripError: null,
          spinner: null,
          spinnerError: null,
        },
        allCities: {
          cities: null,
          citiesError: null,
        },
      },
    },
  },
};
export default mockData;
