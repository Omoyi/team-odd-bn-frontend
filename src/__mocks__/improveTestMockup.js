export const improveTestMockup = {
  handleInputsAction: jest.fn(),
  getStatsAction: jest.fn(),
  changePageNo: jest.fn(),
  getStats: {
    getStatsInput: {
      to: '2020-01-31',
    },
    tripStatsCounter: {
      onewayTripStats: {
        tripsCounter: 2,
      },
      multiTripStats: {
        tripsCounter: 1,
      },
      roundTripStats: {
        tripsCounter: 3,
      },
    },
  },
};
export const searchTripStatisticsProps = {
  stateObject: {
    trips: {
      getStats: {
        getStatsData: {
          userTrips: [],
        },
        getStatsInput: '2020-01-31',
        tripStatsCounter: {},
      },
    },
  },
  getStatsAction: jest.fn(),
  profileError: {
    error: true,
    message: 'You have provided an invalid token',
  },
  history: {
    push: jest.fn(),
  },
};
export const SearchTripStatsViewProps = {
  pageNo: 0,
  itemsPerPage: 1,
  getStats: {
    getStatsInput: {
      tripType: 1,
    },
    tripStatsCounter: {
      onewayTripStats: {
        userTrips: [
          {
            originId: 1,
            destinationId: 2,
            createdAt: '2020-01-28 12:26:00.452+02',
            tripRequestId: 1,
          },
        ],
      },
      roundTripStats: {
        userTrips: [
          {
            originId: 1,
            destinationId: 2,
            createdAt: '2020-01-28 12:26:00.452+02',
            tripRequestId: 1,
          },
        ],
      },
      multiTripStats: {
        userTrips: [
          {
            originId: 1,
            destinationId: 2,
            createdAt: '2020-01-28 12:26:00.452+02',
            tripRequestId: 1,
          },
        ],
      },
    },
  },
};
