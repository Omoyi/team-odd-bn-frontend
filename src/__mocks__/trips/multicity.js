const multicityProps = {
  multicityTripData: {
    data: [
      {
        originId: 1,
        destinationId: 2,
        reason: 'I am reason',
        startDate: '2020-02-02',
        returnDate: '2020-04-04',
      },
    ],
    error: '',
    message: 'Trip requested successfully',
  },
  viewProfile: {
    spinner: true,
  },
  addMoreTrip: jest.fn(),
  editMultipleTrip: jest.fn(),
  removeMoreTrip: jest.fn(),
  sendMulticityTrip: jest.fn(),
  updateSpinnerStatus: jest.fn(),
  tripAction: {
    cities: {
      data: ['tripA', 'tripB'],
    },
    tripIndex: 2,
    removeTrip: jest.fn(),
    onHandleInputChange: jest.fn(),
    data: ['A', 'B'],
    itemRequests: {
      originId: 1,
      destinationId: 2,
      reason: '',
      startDate: '',
      returnDate: '',
    },
  },
  getCitiesAction: jest.fn(),
  history: {
    push: jest.fn(),
  },
};

export const checkInputData = {
  payload: {
    name: 'originId',
    value: 2,
  },
  payload2: {
    name: 'destinationId',
    value: 2,
  },
  payload3: {
    fekeName: 'destinationId',
    fekeValue: 2,
  },
};

export const multicityTripAction = {
  editInputForm: {
    type: 'FEKE_ACTION',
    payload: 'I am data',
  },
  addMoreButton: [
    {
      type: 'ADD_TRIP_REQUEST_MULTIPLE_DATA_SUCCESS',
      payload: {
        originId: '',
        destinationId: '',
        startDate: '',
        returnDate: '',
        reason: '',
      },
    },
  ],
  removeMoreButton: [
    {
      type: 'REMOVE_TRIP_REQUEST_MULTIPLE_DATA_SUCCESS',
      payload: 2,
    },
  ],
  editMultipleTrip: [
    {
      type: 'EDIT_TRIP_REQUEST_MULTIPLE_DATA_SUCCESS',
      payload: { name: 'reason', value: 'why' },
    },
  ],
  multicityObjectData: {
    itinerary: [
      {
        originId: '',
        destinationId: '',
        startDate: '',
        returnDate: '',
        reason: '',
      },
    ],
  },
  expectedMulticityReturn: [
    {
      type: 'SEND_MULTIPLE_DATA_SUCCESS',
      payload: { message: 'Created successfully' },
    },
  ],
  currentMulticityRequest: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity',
      },
    ],
    error: [],
  },
  expectedEditedMulticityRequest: [
    {
      originId: 3,
      destinationId: '6',
      startDate: '2020-12-12',
      returnDate: '2021-12-12',
      reason: 'You should try multicity'
    },
    {
      originId: 9,
      destinationId: '2',
      startDate: '2022-12-12',
      returnDate: '2025-12-12',
      reason: 'You should again try multicity',
    },
  ],
  multicityInputDataPayload: {
    name: 'originId',
    value: 9,
    tripIndex: 1,
  },
  addNewTripRequestResult: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity',
      },
      {
        originId: '',
        destinationId: '',
        startDate: '',
        returnDate: '',
        reason: '',
      },
    ],
    error: [],
  },
  removeTripdReducer: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity',
      },
    ],
    error: [],
  },
  removeReducerAction:
  {
    type: 'REMOVE_TRIP_REQUEST_MULTIPLE_DATA_SUCCESS',
    payload: 0,
  },
  removedDataNewObject: {
    data: [
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity'
      },
    ],
    error: [],
  },
  editSpecificTripRequest: {
    type: 'EDIT_TRIP_REQUEST_MULTIPLE_DATA_SUCCESS',
    payload: {
      tripIndex: 1,
      value: 'i am reason',
      name: 'reason',
    },
  },
  editSpecificTripRequestReason: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'i am reason',
      },
    ],
    error: [],
  },
  successfullyMessage: {
    type: 'SEND_MULTIPLE_DATA_SUCCESS',
    payload: {
      message: 'successfuly created',
    },
  },
  successfullyTripRequestSaved: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity',
      },
    ],
    error: [],
    message: 'successfuly created',
  },
  errorTripRequestMessage: {
    type: 'SEND_MULTIPLE_DATA_ERROR',
    payload: {
      message: 'server is down',
    },
  },
  erroTripRequestSaved: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity',
      },
    ],
    error: 'server is down',
  },
};


export default multicityProps;
