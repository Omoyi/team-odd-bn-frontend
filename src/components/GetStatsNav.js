import React, { Component } from 'react';
import '../assets/css/table.scss';
import propTypes from 'prop-types';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { getStatsAction, handleInputsAction, updateTripCounter } from '../redux/actions/tripsActions/statsTripActions';
import '../assets/css/statsTrip.scss';
import {changePageNo} from '../redux/actions/PaginationAction'
class StatsNav extends Component {
  state = {
    styleButtonOneway: '#34C6F3',
    styleButtonRoundTrip:'white',
    styleButtonMulti:'white',
  }
  render() {
    const {
      getStatsAction, getStats, handleInputsAction,changePageNo
    } = this.props;    
    const { onewayTripStats, multiTripStats, roundTripStats} = getStats.tripStatsCounter;
    return (
      <>
        <div className="statNavContainer">
          <div className="statsTitle">Trip stats</div>
          <div className="dateContainer">
            <div>
              <label>From</label>
              <input
                type="date"
                required
                onChange={(e) => handleInputsAction({ from: e.target.value })}
              />
            </div>
            <div>
              <label>To</label>
              <input
                type="date"
                required
                onChange={(e) => handleInputsAction({ to: e.target.value })}
              />
            </div>
          </div>
          <div className="headButttons">
            <div  style={{ backgroundColor: this.state.styleButtonOneway }} className="oneWayLink">
              <NavLink
                ref="linkToTrip"
                activeClassName="active"
                to="/trips/stats"
                onClick={() => {
                const tripType = 1;
                changePageNo(0);
                  handleInputsAction({ tripType, date: getStats.getStatsInput });
                  this.setState({
                    styleButtonOneway: '#34C6F3',
                    styleButtonRoundTrip:'white',
                    styleButtonMulti:'white',
                  })
                }}
              >
                One Way Trip (
                {onewayTripStats&&onewayTripStats.tripsCounter}
                )
              </NavLink>
            </div>
            <div style={{ backgroundColor: this.state.styleButtonRoundTrip }} className="roundTrip">
              <NavLink
                ref="linkToTrip"
                activeClassName="active"
                to="/trips/stats"
                onClick={() => {
                  const tripType = 2;
                  changePageNo(0);
                  handleInputsAction({ tripType, date: getStats.getStatsInput });
                  this.setState({
                    styleButtonOneway: 'white',
                    styleButtonRoundTrip:'#34C6F3',
                    styleButtonMulti:'white',
                  })
                }}
              >
                Round Trip (
                  {roundTripStats&&roundTripStats.tripsCounter}
                )
              </NavLink>
            </div>
            <div style={{ backgroundColor: this.state.styleButtonMulti }}>
              <NavLink
                ref="linkToTrip"
                activeClassName="active"
                to="/trips/stats"
                onClick={() => {
                  const tripType = 3;
                  changePageNo(0);
                  handleInputsAction({tripType, date: getStats.getStatsInput });
                  this.setState({
                    styleButtonOneway: 'white',
                    styleButtonRoundTrip:'white',
                    styleButtonMulti:'#34C6F3',
                  })
                }}
              >
                Multi City (
                  {multiTripStats&&multiTripStats.tripsCounter}
                )
              </NavLink>
            </div>
          </div>
        </div>
        <div className="searchBtn">
          <button onClick={() => getStatsAction(getStats.getStatsInput)}>
            Search
          </button>
        </div>
      </>
    );
  }
}
const mapStateToProps = (state) => {
  return { getStats: state.trips.getStats };
};
const actions = { getStatsAction, handleInputsAction, updateTripCounter,changePageNo };
export default connect(mapStateToProps, actions)(StatsNav);
