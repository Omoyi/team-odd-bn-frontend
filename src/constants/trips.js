import React from 'react';
import { Link } from 'react-router-dom';
import StatsNav from '../components/GetStatsNav';

export const tableHeads = [
  { className: 'tripTypeNav', attribute: 'Trip Type', key: 'tripType' },
  { className: 'status', attribute: 'Status', key: 'status' },
  { className: 'createdAtNav', attribute: 'Requested On', key: 'Requested On' },
  { className: 'actionNav', attribute: 'Action', key: 'action' },
];
export const navs = [
  {
    className: 'tableHeader',
    attribute:
  <div className="navRequests">
    <div className="travelRequestHeaders"><h2>My Travel Requests</h2></div>
    <div className="newTrip">
      <Link className="add-newTrip-Link" to="/trips/oneway">Add New Trip</Link>
    </div>
  </div>,
    key: 'tableHeader',
  },

];

export const approvalNavs = [
  { className: 'tableHeader', attribute: <h2>Travel Requests For Approval</h2>, key: 'tableHeader' },

];
export const approvalTableHeads = [
  { attribute: 'First Name', key: 'firstname' },
  { attribute: 'Last Name', key: 'lastname' },
  { attribute: 'Trip Type', key: 'tripType' },
  { attribute: 'Requested On', key: 'requested-on' },
  { attribute: 'Action', key: 'action' },
];

export const searchTableHeads = [

  { className: 'tripTypeNav', attribute: 'Origin', key: 'tripType' },
  { className: 'createdAtNav', attribute: 'Destination', key: 'createdAt' },
  { className: 'status', attribute: 'Date', key: 'status' },
  { className: 'actionNav', attribute: 'Action', key: 'action' },
];
export const searchNavs = [
  { className: 'tableHeader', attribute: <StatsNav />, key: 'tableHeader' },
];
export const newTrip = {
  originId: '',
  destinationId: '',
  reason: '',
  startDate: '',
  returnDate: '',
};
