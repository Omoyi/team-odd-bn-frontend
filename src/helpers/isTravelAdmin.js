import verifyToken from './verifyToken';

// eslint-disable-next-line import/prefer-default-export
export const isTravelAdmin = () => {
  const token = localStorage.getItem('token');
  const user = verifyToken(token);
  return (user.roleId === 4);
};
