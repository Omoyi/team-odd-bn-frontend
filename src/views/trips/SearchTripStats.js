import React from 'react';
import { connect } from 'react-redux';
import CommonTable from './CommonTable';
import Action from '../../components/Action';
import { searchNavs, searchTableHeads } from '../../constants/trips';
import { Paginate } from '../../helpers/Paginate';

const RequestsView = ({ getStats, pageNo, itemsPerPage }) => {
  const { getStatsInput, tripStatsCounter } = getStats;

  let data;
  if (getStatsInput.tripType === 1) {
    data = tripStatsCounter.onewayTripStats && tripStatsCounter.onewayTripStats.userTrips;
  }
  if (getStatsInput.tripType === 2) {
    data = tripStatsCounter.roundTripStats && tripStatsCounter.roundTripStats.userTrips;
  }
  if (getStatsInput.tripType === 3) {
    data = tripStatsCounter.multiTripStats && tripStatsCounter.multiTripStats.userTrips;
  }
  const paginatedData = data && Paginate(data, itemsPerPage);
  const currentPageItems = paginatedData
  && paginatedData.length > 0
    ? { ...paginatedData }[pageNo]
    : [];

  const entities = data && data.map((request) => [
    { className: 'tripType', attribute: request.originId, key: 'tripType' },
    { className: 'createdAt', attribute: request.destinationId, key: 'createdAt' },
    { className: 'createdAt', attribute: request.createdAt && request.createdAt.slice(0, 10), key: 'date' },
    { className: 'action', attribute: <Action background="#34c6f3" url={`/requests/${request.tripRequestId}`} action="View" />, key: 'action' },
  ]);

  return (
    <CommonTable
      data={paginatedData}
      navs={searchNavs}
      tableHeads={searchTableHeads}
      entities={entities || []}
    />
  );
};


export const mapStateToProps = (state) => ({
  getStats: state.trips.getStats,
  pageNo: state.pagination.pageNo,
  itemsPerPage: state.pagination.itemsPerPage,
});


export default connect(mapStateToProps, null)(RequestsView);
