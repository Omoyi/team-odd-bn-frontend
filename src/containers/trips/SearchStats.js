import React, { Component } from 'react';
import { connect } from 'react-redux';
import '../../assets/css/table.scss';
import SearchTripStats from '../../views/trips/SearchTripStats';
import Dashboard from '../../views/Dashboard/sidebar';
import { getStatsAction, updateTripCounter } from '../../redux/actions/tripsActions/statsTripActions';
import verifyToken from '../../helpers/verifyToken';
import tokenExist from '../../helpers/tokenExist';

class Requests extends Component {
  async componentDidMount() {
    if (!verifyToken(tokenExist)) {
      return this.props.history.push('/signin');
    }
    const { getStatsAction } = this.props;
    await getStatsAction({ tripType: '1', from: '2020-01-22', to: '2020-01-26' });


    if (this.props.profileError.error
  || this.props.profileError.message === 'You have provided an invalid token') {
      return this.props.history.push('/signin');
    }
  }

  render() {
    const { getStatsData, getStatsInput, tripStatsCounter } = this.props.stateObject.trips.getStats;
    const data = getStatsData && getStatsData.userTrips;
    return (
      <Dashboard>
        <SearchTripStats data={data} getStatsInput={getStatsInput} tripStatsCounter={tripStatsCounter} />
      </Dashboard>
    );
  }
}

export const mapStateToProps = (state) => ({
  stateObject: state,
});
const actions = {
  getStatsAction,
  updateTripCounter,
};

export default connect(mapStateToProps, actions)(Requests);
