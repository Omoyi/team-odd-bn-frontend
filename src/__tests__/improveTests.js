import React from 'react';
import { shallow } from 'enzyme';
import moxios from 'moxios';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { StatsNav } from '../components/GetStatsNav';
import { Requests } from '../containers/trips/SearchStats';
import { RequestsView as SearchTripStats } from '../views/trips/SearchTripStats';
import { isTravelAdmin } from '../helpers/isTravelAdmin';
import {
  getStatsAction,
  updateSpinnerStatus,
  handleInputsAction,
  updateTripCounter,
} from '../redux/actions/tripsActions/statsTripActions';
import apiCall from '../helpers/apiCall';
import { SearchTripStatsViewProps, improveTestMockup, searchTripStatisticsProps } from '../__mocks__/improveTestMockup';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);


describe('Expect to test and display trip stats with navigation', () => {
  const component = shallow(<StatsNav {...improveTestMockup} />);
  it('Should render successfully', () => {
    expect(component).toMatchSnapshot();
  });
  it('Should simulate on change type from input function', () => {
    const addInput = component.find('input').at(0);
    addInput.simulate('change', { target: { value: '2020-01-31' } });
  });

  it('Should simulate on change type to input function', () => {
    const addInput = component.find('input').at(1);
    addInput.simulate('change', { target: { value: '2020-01-31' } });
  });

  it('Should simulate on click NavLink for oneway to handle input function', () => {
    const addNavLink = component.find('NavLink').at(0);
    addNavLink.simulate('click');
  });

  it('Should simulate on click NavLink for round trip to handle input function', () => {
    const addNavLink = component.find('NavLink').at(1);
    addNavLink.simulate('click');
  });

  it('Should simulate on click NavLink for multicity to handle input function', () => {
    const addNavLink = component.find('NavLink').at(2);
    addNavLink.simulate('click');
  });

  it('Should simulate on click button search trip ', () => {
    const addButton = component.find('button');
    addButton.simulate('click');
  });
});

describe('Expect to search stats nav', () => {
  const component = shallow(<Requests {...searchTripStatisticsProps} />);
  it('Should render successfully', () => {
    expect(component).toMatchSnapshot();
  });
});

describe('Expect to search stats nav', () => {
  it('Should search trips oneway statics viewed', () => {
    const viewSearchOneWayTrips = SearchTripStats(SearchTripStatsViewProps);
    expect(viewSearchOneWayTrips).toBeTruthy();
  });
  it('Should search trips roundTrips statics viewed', () => {
    SearchTripStatsViewProps.getStats.getStatsInput.tripType = 2;
    SearchTripStatsViewProps.getStats.tripStatsCounter.onewayTripStats = '';
    const viewSearchOneWayTrips = SearchTripStats(SearchTripStatsViewProps);
    expect(viewSearchOneWayTrips).toBeTruthy();
  });
  it('Should search trips roundTrips statics viewed', () => {
    SearchTripStatsViewProps.getStats.getStatsInput.tripType = 3;
    SearchTripStatsViewProps.getStats.tripStatsCounter.onewayTripStats = '';
    SearchTripStatsViewProps.getStats.tripStatsCounter.roundTripStats = '';
    const viewSearchOneWayTrips = SearchTripStats(SearchTripStatsViewProps);
    expect(viewSearchOneWayTrips).toBeTruthy();
  });
});

/* helpers/isTravelAdmin */
describe('Expect to verify travel admin', () => {
  isTravelAdmin();
});

/* src/redux/actions/tripsActions/statsTripActions.js */
const inputSearch = {
  from: '2020-01-31',
  to: '2020-05-31',
  tripType: undefined,
};

describe('Expect to mock error and reject to view accommodation', () => {
  beforeEach(() => {
    moxios.install(apiCall);
  });

  afterEach(() => {
    moxios.uninstall(apiCall);
  });

  it('Should mock search request and return error response', async () => {
    const errorResp = {
      status: 400,
      response: {
        data: {
          message: 'No trips search found',
        },
      },
    };

    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.reject(errorResp);
    });

    const store = mockStore({});
    await store.dispatch(getStatsAction(inputSearch)).catch((error) => {
      expect(error.status).toEqual(errorResp.status);
      expect(error.response.message).toEqual(errorResp.response.message);
    });
  });
});

describe('Expect to test all trips helper function', () => {
  it('Should return updateSpinnerStatus payload', () => {
    const result = updateSpinnerStatus(false);
    expect(result).toEqual({
      type: 'UPDATE_SPINNER_STATUS',
      spinner: false,
    });
  });

  it('Should return handleInputsAction payload', () => {
    const result = handleInputsAction('feke data');
    expect(result).toEqual({
      type: 'GET_STATS_INPUT',
      payload: 'feke data',
    });
  });

  it('Should return updateTripCounter payload', () => {
    const result = updateTripCounter('feke data');
    expect(result).toEqual({
      type: 'TRIP_COUNTER_INPUT',
      payload: 'feke data',
    });
  });
});
