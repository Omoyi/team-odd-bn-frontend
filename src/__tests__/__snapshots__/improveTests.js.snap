// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to search stats nav Should render successfully 1`] = `ShallowWrapper {}`;

exports[`Expect to test and display trip stats with navigation Should render successfully 1`] = `ShallowWrapper {}`;
